### DISCUSSION TITLE
Prisma doesn't support GraphQL Subscriptions ?

### DISCUSSION THREAD

**Discussion by AbhijithGanesh** [prisma/prisma#19248] [Q&A] [answered]
### Bug description

I have currently configured [Apollo Server](https://www.apollographql.com/docs) as my GraphQL client and Next+Apollo for the server. I have written the logic to query for data (mutate) and send it as a subscription! Unfortunately, whenever I try accessing the subscription, I get this error

<hr/>

## Resolver

```ts
  Mutation: {
    createPost(parent, args, ctxt, info) {
      const { title } = args;
      let body = {
        id: 1,
        title: "ABhijithG",
      };
      pubSub.publish("newPost", { body });
      return body;
    },
  },
  Subscription: {
    recieveUpdates: {
      resolve: (payload) => payload.newPost,
      subscribe: () => pubSub.asyncIterator("newPost"),
    },
  },
```

## Schema/Type Defs
```ts
import { gql } from "@apollo/client";

export const typeDefs = gql`
  type Post {
    id: Int!
    title: String!
  }

  type Query {
    viewer: [Post]
  }

  type Mutation {
    createPost(title: String!): Post!
  }

  type Subscription {
    recieveUpdates: Post!
  }
`;

```

![image](https://github.com/prisma/prisma/assets/67182544/c29b13eb-e80c-4e5d-a748-0638f99a22ba)

[Repository](https://github.com/AbhijithGanesh/nextjs-subscripitons-poc)

### How to reproduce

1. clone the repository
2. Run `yarn turbo`

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: Fedora 38
- Prisma: Latest (4.14)
- Next: Latest (13)
- Apollo (Server: 4.*)

### Prisma Version

```

```
4.14

**Comment by nurul3101** [comment, marked as answer]
Hello @AbhijithGanesh 👋 

Can you share how are you querying your data?

The issue would be on how you are initialising and using the PrismaClient Instance.
Here's a related GitHub Issue that describes various solutions to this error:
- https://github.com/prisma/prisma/issues/6219

I would recommend you to have a look at the solutions in the above GitHub Issue.

**Reply by nurul3101** [reply]
@AbhijithGanesh If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
I'll mark this discussion as closed. Please feel free to reopen if needed.
